lib/cc-reno.c

```diff
@@ -39,6 +40,7 @@ static void reno_on_acked(quicly_cc_t *cc, const quicly_loss_t *loss, uint32_t b
         return;
     }
     /* Congestion avoidance. */
+    cc->pacer_multiplier = QUICLY_PACER_CALC_MULTIPLIER(1.2);
```

Comment:
Actually, it seems that 1.2x in congestion avoidance mode is sub-optimal, sometimes leading to underutilization of the bottleneck bandwidth.

When a loss is observed, CWND is immediately halved (if BETA is 0.5), or reduced to 70%. But SRTT does not get reduced immediately; in fact, it might take multiple round trips for srtt to adopt to the new situation.

That means that immediately after the loss event, bandwidth estimate that the pacer relies on (i.e., CWND / srtt) can become 60% of the delivery rate (i.e, 0.5 * 1.2), or 84% (0.7 * 1.2).

Using `min(latest_rtt, srtt)` mitigates this issue to some extend but not completely. Because when the sender resumes sending, latest_rtt will be no more than 0.75 compared to what is observed before the loss event. 0.75 * 1.2 is 0.9 (i.e., below 1).

Ignoring jitters, the correct value to be used here is no smaller than `1 / beta`.

Reply:
Discussed with @janaiyengar and agreed to start with using 2x in both slow start and congestion avoidance phase.

4ecb822 makes the change.

As the constant of `2` only appears once, it is easy to extend the logic to support more than one multipliers based on the state of the connection.